holography/interferometric_raster_track.py

```diff
@@ -0,0 +1,190 @@
+            if not session.cbf.fengine.inputs:
+                raise RuntimeError("Failed to get correlator input labels, "
+                                   "cannot set the F-engine gains")
+            for inp in session.cbf.fengine.inputs:
+                session.cbf.fengine.req.gain(inp, opts.reset_gain)
+                user_logger.info("F-engine %s gain set to %g",
+                                 inp, opts.reset_gain)
+
+        session.standard_setup(**vars(opts))
+        if opts.fft_shift is not None:
+            session.cbf.fengine.req.fft_shift(opts.fft_shift)
+        session.capture_start()
+
+        start_time = time.time()
+
+        targets_observed = []
+        # Keep going until the time is up
+        target_total_duration = [0.0] * len(targets)
+        keep_going = True
+        
+        last_lunar_scan = -np.inf
+        last_cal_visit = -np.inf
+        while keep_going:
+            keep_going = (opts.max_duration is not None) and opts.repeat
+            targets_before_loop = len(targets_observed)
+            # Iterate through source list, picking the next one that is up
+            target_list = list(targets.filter(['~bpcal', '~gaincal']))
+            n = 0
+            while n < len(target_list):
+                # Cut the track short if time ran out
+                def _time_left(opts, start_time, track_duration):
+                    """ compute duration based on time remaining """
+                    if track_duration is not None:
+                        time_left = opts.max_duration - (time.time() - start_time)
+                        if time_left <= 0.:
+                            user_logger.warning("Maximum duration of %g seconds "
+                                                "has elapsed - stopping script",
+                                                opts.max_duration)
+                            return 0, False
+                        duration = min(track_duration, time_left)
+                        return duration, True
+
+                def _go_observe(session, target, targets_observed, target_total_duration, duration):
+                    session.label('track')
+                    if session.track(target, duration=duration):
+                        targets_observed.append(target.description)
+                        target_total_duration[n] += duration
+                # select between calibrator, moon or target
+                if time.time() - last_lunar_scan > opts.moon_repeat:
+                    # time to observe moon
+                    last_lunar_scan = time.time()
+                    for lunar_track in range(opts.moon_nsnapshots_per_track):
+                        duration, keep_going = _time_left(opts, start_time, opts.moon_snapshot_duration)
+                        if not keep_going: 
+                            #user_logger.info("DEBUG: Out of time. Stopping")
+                            break
+                        observer = ephem.Observer()
```

Comment:
I realise that this is a stale PR, but I want to resolve the review request 😂 

For starters, I would not use `ephem` directly when `katpoint` provides exactly the same functionality:
```Python
moon = kat.sources['Moon']
moon_radec = moon.radec(time.time() + duration / 2.0)
target = katpoint.construct_radec_target(*moon_radec)
```